website/docs/data-display/stacked-horizontal-bar/examples/legend.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import {
+  Plot,
+  StackBar,
+  YAxis,
+  XAxis,
+  ChartLegend,
+  LegendItem,
+  makeDataHintsContainer,
+} from '@semcore/ui/d3-chart';
+import { scaleLinear, scaleBand } from 'd3-scale';
+import { Flex } from '@semcore/ui/flex-box';
+import resolveColor from '@semcore/ui/utils/lib/color';
+import Card from '@semcore/ui/card';
+
+const barColors = {
+  1: resolveColor('blue-300'),
+  2: resolveColor('blue-200'),
+  3: resolveColor('yellow-200'),
+};
+
+const dataHints = makeDataHintsContainer();
+
+export default () => {
+  const MARGIN = 40;
+  const width = 500;
+  const height = 300;
+
+  const xScale = scaleLinear()
+    .range([MARGIN * 2, width - MARGIN])
+    .domain([0, 20]);
+
+  const yScale = scaleBand()
+    .range([height - MARGIN, MARGIN])
+    .domain(data.map((d) => d.category))
+    .paddingInner(0.4)
+    .paddingOuter(0.2);
+
+  const [legendItems, setLegendItems] = React.useState(
+    Object.keys(data[0])
+      .filter((name) => name !== 'category')
+      .reduce<LegendItem[]>((res, item) => {
```

Comment:
Is it just a rocket-science form of `.map`? 😁

Reply:
it was copy-paste from some another place, sorry. fixed